### PR TITLE
feat(wallet): NFT Grid View

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Built out NFT GridView

Introduces 3 new filter options

- Filter by Balance (low to high)
- Filter by Balance (high to low)
- Filter by NFT's

Resolves <https://github.com/brave/brave-browser/issues/22585>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review is not needed
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://user-images.githubusercontent.com/40611140/182291164-dca7c3f6-da57-4260-b41f-f57279697d5b.mov

### CODE REVIEW

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx [hunk 1]
```diff
@@ -60,18 +69,38 @@ const TokenLists = (props: Props) => {
...
 
-  // memos
-  const nonFungibleTokenList = React.useMemo(
-    () => filteredAssetList.filter(({ asset }) => asset.isErc721),
-    [filteredAssetList]
-  )
+  // methods
+  const {
+    selectedAssetFilter
+  } = useSelector(({ wallet }: { wallet: WalletState }) => wallet)
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx:75]:
```suggestion
  const selectedAssetFilter = useSelector(({ wallet }: { wallet: WalletState }) => wallet.selectedAssetFilter)
```

File: components/brave_wallet_ui/components/shared/nft-icon/nft-icon.tsx [hunk 2]
```diff
@@ -25,15 +34,27 @@ export const NftIcon = ({ icon }: AssetIconProps) => {
         payload: icon
       }
       sendMessageToNftUiFrame(nftImageIframeRef.current.contentWindow, command)
+      const loadingCommand: UpdateLoadingMessage = {
+        command: NftUiCommand.UpdateLoading,
+        payload: false
+      }
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/shared/nft-icon/nft-icon.tsx:40]: can be defined outside of component

File: components/brave_wallet_ui/constants/types.ts [hunk 3]
```diff
@@ -695,3 +696,14 @@ export type OriginInfo = {
   origin: string
   eTldPlusOne: string
 }
+
+export type AssetFilterOptions =
```
Comment (josheleonard) [inline components/brave_wallet_ui/constants/types.ts:700]: How about `AssetFilterOptionIds`?

File: components/brave_wallet_ui/constants/types.ts [hunk 4]
```diff
@@ -695,3 +696,14 @@ export type OriginInfo = {
...
+  | 'allAssets'
+  | 'nfts'
+  | 'highToLow'
+  | 'lowToHigh'
+
+export interface AssetFilterOptionObjectType {
```
Comment (josheleonard) [inline components/brave_wallet_ui/constants/types.ts:706]: How about `AssetFilterOption`?

File: components/brave_wallet_ui/nft/components/nft-content/nft-content.tsx [hunk 5]
```diff
@@ -33,15 +33,16 @@ export const NftContent = (props: Props) => {
...
-        : <>
-          {isLoading &&
-            <LoadingOverlay isLoading={isLoading}>
-              <LoadIcon />
-            </LoadingOverlay>
+      {isLoading ? (
+        <LoadingOverlay isLoading={isLoading}>
+          <LoadIcon />
+        </LoadingOverlay>
+      ) : (
```
Comment (josheleonard) [inline components/brave_wallet_ui/nft/components/nft-content/nft-content.tsx:40]: Might be cleaner to write a plain `if` for ``isLoading` and return early

File: components/brave_wallet_ui/options/asset-filter-options.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { AssetFilterOptionObjectType } from '../constants/types'
```
Comment (josheleonard) [inline components/brave_wallet_ui/options/asset-filter-options.ts:1]: License

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/common/actions/wallet_actions.ts [hunk 7]
```diff
@@ -128,6 +129,7 @@ export const setTransactionProviderError = createAction<SetTransactionProviderEr
 export const setSelectedCoin = createAction<BraveWallet.CoinType>('setSelectedCoin')
 export const setDefaultNetworks = createAction<BraveWallet.NetworkInfo[]>('setDefaultNetworks')
 export const setSelectedNetworkFilter = createAction<BraveWallet.NetworkInfo>('setSelectedNetworkFilter')
+export const setSelectedAssetFilterItem = createAction<AssetFilterOptionObjectType>('setSelectedAssetFilterItem')
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/actions/wallet_actions.ts]: Yes! Will eventually be a user pref that we will need to manage here. Same thing for `selectedNetworkFilter`

Here are the issue for both

https://github.com/brave/brave-browser/issues/24387

https://github.com/brave/brave-browser/issues/24388

File: components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/asset-filter-item.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+  BigCheckMark
+} from '../network-filter-selector/style'
+
+export interface Props {
+  selectedAssetFilterItem: AssetFilterOptionObjectType
+  assetFilterItem: AssetFilterOptionObjectType
+  onSelectAssetFilterItem: (assetFilterItem: AssetFilterOptionObjectType) => void
+}
+
+function AssetFilterItem (props: Props) {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/asset-filter-item.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/asset-filter-item.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+// Styled Components
+import {
+  NetworkItemButton,
+  NetworkName,
+  NetworkItemWrapper,
+  BigCheckMark
+} from '../network-filter-selector/style'
+
+export interface Props {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/asset-filter-item.tsx]: Not sure why that was being exported, but it doesn't look like it. Removed!

File: components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/asset-filter-item.tsx [hunk 10]
```diff
@@ -0,0 +1,43 @@
+import * as React from 'react'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/asset-filter-item.tsx:6]: Added!

File: components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/index.tsx [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+import {
+  StyledWrapper,
+  DropDown,
+  DropDownButton,
+  DropDownIcon,
+  LeftSide,
+  ClickAwayArea
+} from '../network-filter-selector/style'
+
+function AssetFilterSelector () {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/index.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/index.tsx [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+  const [showAssetFilter, setShowAssetFilter] = React.useState<boolean>(false)
+
+  const {
+    selectedAssetFilter
+  } = useSelector(({ wallet }: { wallet: WalletState }) => wallet)
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/index.tsx]: Updated!

File: components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/index.tsx [hunk 13]
```diff
@@ -0,0 +1,76 @@
+import * as React from 'react'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/asset-filter-selector.tsx/index.tsx]: Renamed!

File: components/brave_wallet_ui/components/desktop/index.ts [hunk 14]
```diff
@@ -30,6 +30,7 @@ import SwapTooltip from './swap-tooltip'
 import WithHideBalancePlaceholder from './with-hide-balance-placeholder'
 import NetworkFilterSelector from './network-filter-selector'
 import { CryptoView, PortfolioAsset, PortfolioOverview } from './views'
+import AssetFilterSelector from './asset-filter-selector.tsx'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/index.ts]: Removed!

File: components/brave_wallet_ui/components/desktop/network-filter-selector/style.ts [hunk 15]
```diff
@@ -12,13 +16,13 @@ export const StyledWrapper = styled.div`
   position: relative;
 `
 
-export const DropDownButton = styled(WalletButton)`
+export const DropDownButton = styled(WalletButton) <StyleProps>`
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/network-filter-selector/style.ts]: Done!

File: components/brave_wallet_ui/components/desktop/network-filter-selector/style.ts [hunk 16]
```diff
@@ -3,6 +3,10 @@ import { WalletButton } from '../../shared/style'
 import { CaratStrongDownIcon } from 'brave-ui/components/icons'
 import CheckMark from '../../../assets/svg-icons/big-checkmark.svg'
 
+interface StyleProps {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/network-filter-selector/style.ts]: Removed!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/index.tsx [hunk 17]
```diff
@@ -0,0 +1,35 @@
+import * as React from 'react'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/index.tsx]: Renamed!

[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/index.tsx]: Added!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/index.tsx [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+import {
+  StyledWrapper
+} from './style'
+
+interface Props {
+  erc721TokenList: UserAssetInfoType[]
+  onSelectAsset: (asset: BraveWallet.BlockchainToken | undefined) => () => void
+}
+
+const NFTGridView = (props: Props) => {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/index.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view-item.tsx [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+  IconWrapper,
+  DIVForClickableArea
+} from './style'
+
+interface Props {
+  token: UserAssetInfoType
+  onSelectAsset: () => void
+}
+
+const NFTGridViewItem = (props: Props) => {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view-item.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view-item.tsx [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+
+  return (
+    <NFTButton
+      onClick={onSelectAsset}
+    >
+      <IconWrapper>
+        <DIVForClickableArea />
+        <NftIcon icon={remoteImage} responsive={true} />
+      </IconWrapper>
+      <NFTText>{token.asset.name} {'#' + new Amount(token.asset.tokenId).toNumber()}</NFTText>
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view-item.tsx:45]: `tokenId` needs to be converted from `hex` value to `number`.
`Amount` can be utilized here.

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view-item.tsx [hunk 21]
```diff
@@ -0,0 +1,42 @@
+import * as React from 'react'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view-item.tsx:6]: Added!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/style.ts [hunk 22]
```diff
@@ -0,0 +1,65 @@
+import styled from 'styled-components'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/style.ts:6]: Added!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/style.ts [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+  flex-direction: column;
+  cursor: pointer;
+  outline: none;
+  background: none;
+  border: none;
+  margin: 0px;
+  padding: 0px;
+  &:last-child {
+    margin-right: 0px;
+}
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/style.ts]: Fixed!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx]: Updated!

File: components/brave_wallet_ui/components/shared/nft-icon/nft-icon.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/shared/nft-icon/nft-icon.tsx]: Done!

File: components/brave_wallet_ui/constants/types.ts (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/constants/types.ts]: Changed!

File: components/brave_wallet_ui/constants/types.ts (hunk 4 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/constants/types.ts]: Changed!

File: components/brave_wallet_ui/nft/components/nft-content/nft-content.tsx (hunk 5 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/nft/components/nft-content/nft-content.tsx]: Good call! Done!

File: components/brave_wallet_ui/options/asset-filter-options.ts (hunk 6 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/options/asset-filter-options.ts]: Added!

File: components/brave_wallet_ui/common/actions/wallet_actions.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/common/actions/wallet_actions.ts]: Wouldn't we then be able to query this pref's value directly with the ApiProxy?

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx [hunk 24]
```diff
@@ -60,18 +69,36 @@ const TokenLists = (props: Props) => {
...
+        const aFiatBalance = computeFiatAmount(aBalance, a.asset.symbol, a.asset.decimals)
+        return aFiatBalance.toNumber() - bFiatBalance.toNumber()
+      })
+    }
+    return fungibleTokens
+  }, [filteredAssetList, selectedAssetFilter, computeFiatAmount])
 
-  // methods
   // This filters a list of assets when the user types in search bar
   const onFilterAssets = React.useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx:103]: This should be refactored. onFilterAssets should set the search value to state, then a memo should run to filter the list when that value changes

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx [hunk 25]
```diff
@@ -60,18 +69,36 @@ const TokenLists = (props: Props) => {
...
+    }
+    if (selectedAssetFilter.id === 'lowToHigh') {
+      return fungibleTokens.sort(function (a, b) {
+        const aBalance = a.assetBalance
+        const bBalance = b.assetBalance
+        const bFiatBalance = computeFiatAmount(bBalance, b.asset.symbol, b.asset.decimals)
+        const aFiatBalance = computeFiatAmount(aBalance, a.asset.symbol, a.asset.decimals)
+        return aFiatBalance.toNumber() - bFiatBalance.toNumber()
+      })
+    }
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx:98]:
```suggestion
if (
    selectedAssetFilter.id === 'highToLow' ||
    selectedAssetFilter.id === 'lowToHigh'
  ) {
    return fungibleTokens.sort(function (a, b) {
      const aBalance = a.assetBalance
      const bBalance = b.assetBalance
      const bFiatBalance = computeFiatAmount(bBalance, b.asset.symbol, b.asset.decimals)
      const aFiatBalance = computeFiatAmount(aBalance, a.asset.symbol, a.asset.decimals)
      return selectedAssetFilter.id === 'highToLow' 
        ? bFiatBalance.toNumber() - aFiatBalance.toNumber()
        : aFiatBalance.toNumber() - bFiatBalance.toNumber()
    })
  }
```

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx (hunk 24 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx (hunk 25 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/index.tsx]: I like it! Changed!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view.tsx [hunk 26]
```diff
@@ -0,0 +1,38 @@
...
+import { NFTGridViewItem } from './nft-grid-view-item'
+
+// Styled Components
+import {
+  StyledWrapper
+} from './style'
+
+interface Props {
+  nonFungibleTokens: UserAssetInfoType[]
+  onSelectAsset: (asset: BraveWallet.BlockchainToken | undefined) => () => void
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view.tsx:21]:
```suggestion
  onSelectAsset: (asset: BraveWallet.BlockchainToken | undefined) => void
```

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view.tsx [hunk 27]
```diff
@@ -0,0 +1,38 @@
...
+export const NFTGridView = (props: Props) => {
+  const { nonFungibleTokens, onSelectAsset } = props
+
+  return (
+    <StyledWrapper>
+      {nonFungibleTokens.map((token: UserAssetInfoType) =>
+        <NFTGridViewItem
+          key={`${token.asset.tokenId}-${token.asset.contractAddress}`}
+          token={token}
+          onSelectAsset={onSelectAsset(token.asset)}
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view.tsx:33]:
```suggestion
          onSelectAsset={() => onSelectAsset(token.asset)}
```

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 28]
```diff
@@ -0,0 +1,185 @@
...
+  FilterTokenRow
+} from '../../style'
+
+interface Props {
+  tokenPrices: BraveWallet.AssetPrice[]
+  defaultCurrencies: DefaultCurrencies
+  userAssetList: UserAssetInfoType[]
+  hideBalances: boolean
+  networks: BraveWallet.NetworkInfo[]
+  onSelectAsset: (asset: BraveWallet.BlockchainToken | undefined) => () => void
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:52]:
```suggestion
  onSelectAsset: (asset: BraveWallet.BlockchainToken | undefined) =>  void
```

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 29]
```diff
@@ -0,0 +1,185 @@
...
+        <NetworkFilterSelector />
+        <AssetFilterSelector />
+      </FilterTokenRow>
+      {selectedAssetFilter.id !== 'nfts' ? (
+        <>
+          {sortedAssetList.map((item) =>
+            <PortfolioAssetItem
+              spotPrices={tokenPrices}
+              defaultCurrencies={defaultCurrencies}
+              action={onSelectAsset(item.asset)}
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:142]:
```suggestion
              action={() => onSelectAsset(item.asset)}
```

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx [hunk 30]
```diff
@@ -0,0 +1,185 @@
...
+          {nonFungibleTokens.length !== 0 &&
+            <>
+              <Spacer />
+              <DividerText>{getLocale('braveWalletTopNavNFTS')}</DividerText>
+              <SubDivider />
+              {nonFungibleTokens.map((item) =>
+                <PortfolioAssetItem
+                  spotPrices={tokenPrices}
+                  defaultCurrencies={defaultCurrencies}
+                  action={onSelectAsset(item.asset)}
```
Comment (josheleonard) [inline components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx:159]:
```suggestion
                  action={() => onSelectAsset(item.asset)}
```

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view.tsx (hunk 26 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view.tsx (hunk 27 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx (hunk 28 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx (hunk 29 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx]: Done!

File: components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx (hunk 30 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx]: Done!

**Review by josheleonard — APPROVED** [review]

**Review by muliswilliam — APPROVED** [review]

**Pull request merged**